src/model/level/mod.rs

```diff
@@ -347,16 +382,273 @@ impl Display for Password {
         }
     }
 }
+
+/// Struct representing partial levels. These are returned to
+/// [`LevelsRequest`]s and only
+/// contain metadata
+/// on the level.
+///
+/// ## GD Internals:
+/// The Geometry Dash servers provide lists of partial levels via the
+/// `getGJLevels` endpoint.
+///
+/// ### Unmapped values:
+/// + Index `8`: Index 8 is a boolean value indicating whether the level has a
+/// difficulty rating that isn't N/A. This is equivalent to checking if
+/// [`PartialLevel::difficulty`] is unequal to
+/// [`LevelRating::NotAvailable`]
+/// + Index `17`: Index 17 is a boolean value indicating whether
+/// the level is a demon level. This is equivalent to checking if
+/// [`PartialLevel::difficulty`] is the [`LevelRating::Demon`] variant.
+/// + Index `25`: Index 25 is a boolean value indicating
+/// whether the level is an auto level. This is equivalent to checking if
+/// [`PartialLevel::difficulty`] is equal to
+/// [`LevelRating::Auto`].
+///
+/// ### Unprovided values:
+/// These values are not provided for by the `getGJLevels` endpoint and are
+/// thus only modelled in the [`Level`] struct: `4`, `27`,
+/// `28`, `29`, `36`
+///
+/// ### Unused indices:
+/// The following indices aren't used by the Geometry Dash servers: `11`, `16`,
+/// `17`, `20`, `21`, `22`, `23`, `24`, `26`, `31`, `32`, `33`, `34`, `40`,
+/// `41`, `44`
+#[derive(Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct Level<'a, Song, User> {
```

Comment:
Same for the comment on docs, checking from http://docs.gdprogra.me/#/resources/server/level